### PR TITLE
Feature/gatsby5

### PR DESCRIPTION
**Pull request by schabibi1**
To do before merge this PR:
- [ ] Create Gatsby 4 supported branch from the current master

### CODE REVIEW

File: lib/package.json [hunk 1]
```diff
@@ -40,14 +39,14 @@
...
-    "react-dom": "^17.0.2",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
     "start-server-and-test": "^2.0.0",
     "typescript": "^5.1.3",
     "vite": "^4.3.9"
   },
   "peerDependencies": {
-    "gatsby": "^3.0.0 || ^4.0.0"
+    "gatsby": "^3.0.0 || ^4.0.0 || ^5.0.0"
```
Comment (alexjoverm) [inline lib/package.json:49]: As this package depends on `gatsy-source-filesystem@5.7.0` (for Gatsby 5) I think that would make it incompatible with Gatsby 4 - if this is correct, we should just keep `^5.0.0` in the `peerDependencies`

File: lib/tsconfig.json [hunk 2]
```diff
@@ -16,9 +17,38 @@
...
+//     "incremental": true
+//   },
+//   "include": [
+//     "**/*.ts",
+//     "**/*.tsx"
+//   ],
+//   "exclude": [
+//     "node_modules"
+//   ]
+// }
```
Comment (alexjoverm) [inline lib/tsconfig.json:54]: @schabibi1 can we remove this comments?

**Review by alexjoverm — CHANGES_REQUESTED** [review]

File: lib/package.json (hunk 1 above)
Comment (schabibi1) [reply lib/package.json:49]: Somehow, I have `^5.0.0` already included in my local and it seems it was already added last month @alexjoverm

Comment (schabibi1) [reply lib/package.json:49]: Screenshot 2023-07-07 at 14 36 58

File: lib/tsconfig.json (hunk 2 above)
Comment (schabibi1) [reply lib/tsconfig.json]: Done @alexjoverm

File: lib/package.json (hunk 1 above)
Comment (schabibi1) [reply lib/package.json]: Done @alexjoverm

**Pull request merged**